it/repositories/application/FsbTestGroupMongoRepositorySpec.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.application
+
+import config.MicroserviceAppConfig.cubiksGatewayConfig
+import factories.{ DateTimeFactory, UUIDFactory }
+import model.EvaluationResults.{ Green, Red }
+import model.SchemeId
+import model.persisted._
+import reactivemongo.bson.BSONDocument
+import reactivemongo.json.ImplicitBSONHandlers
+import repositories.CollectionNames
+import testkit.MongoRepositorySpec
+
+class FsbTestGroupMongoRepositorySpec extends MongoRepositorySpec with UUIDFactory {
+
+  import ImplicitBSONHandlers._
+
+  val collectionName = CollectionNames.APPLICATION
+  def repository = new FsbTestGroupMongoRepository()
+
+  "save" should {
+    "create new FSB entry in testGroup if it doesn't exist" in {
+      val applicationId = createApplication()
+      val schemeEvaluationResult = SchemeEvaluationResult("GovernmentOperationalResearchService", "Green")
+      repository.save(applicationId, schemeEvaluationResult).futureValue
+      val Some(result) = repository.findByApplicationId(applicationId).futureValue
+      val fsbTestGroup = FsbTestGroup(List(schemeEvaluationResult))
+      result mustBe fsbTestGroup
+    }
+
+    "add to result array if result array already exist" in {
+      val applicationId = createApplication()
+      val schemeEvaluationResult1 = SchemeEvaluationResult("GovernmentOperationalResearchService", "Red")
+      repository.save(applicationId, schemeEvaluationResult1).futureValue
+
+      val schemeEvaluationResult2 = SchemeEvaluationResult("GovernmentSocialResearchService", "Green")
+      repository.save(applicationId, schemeEvaluationResult2).futureValue
+
+      val Some(result) = repository.findByApplicationId(applicationId).futureValue
+      val expectedFsbTestGroup = FsbTestGroup(List(schemeEvaluationResult1, schemeEvaluationResult2))
+      result mustBe expectedFsbTestGroup
+    }
+
+    "not overwrite existing value" in {
+      val applicationId = createApplication()
+      repository.save(applicationId, SchemeEvaluationResult("GovernmentOperationalResearchService", "Green")).futureValue
+
+      intercept[Exception] {
```

Comment:
Because you are throwing a generic exception here, this test is not realllllly valid. You'll need to throw a custom exception and intercept that.